### PR TITLE
Use the core client with explicit version globally

### LINKED ISSUE #49535: Use the core client with version
**Is this a BUG REPORT or FEATURE REQUEST?**:

**What happened**:
When rebase #48976 to fix the conflict with #48847, find that some codes use `clientSet.CoreV1()`, while some codes still use the **Deprecated** `clientSet.Core()`.

**What you expected to happen**:
Replace the **Deprecated** `clientSet.Core()` with `clientSet.CoreV1()`.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by kevin-wangzefeng**
**What this PR does / why we need it**:
As mentioned in #49535 and #50605, we want to have a global replace to use core client with explicit version.

**Which issue this PR fixes**: fixes #49535 

**Special notes for your reviewer**:
The actual type of clientSet.Core() is already the same with clientSet.CoreV1(), so it should be safe replacement.
The places that clientSet.Core() are still in use are identified by IDE "find usages", and changes are made with one time global replace. Hopefully there will be none left after this PR merged.
Let me know if this PR is too big to review, I can split it into some smaller ones.

/cc @kubernetes/sig-api-machinery-pr-reviews 
/cc @k82cn @sttts 

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by sttts** [comment]
/lgtm
/cc @caesarxuchao 
@cheftako Walter, can you approve this on behalf of Daniel?

**Comment by caesarxuchao** [comment]
@kevin-wangzefeng you might want to coordinate with @supereagle. You are working on the same sort of refactoring.

**Comment by supereagle** [comment]
@caesarxuchao I think the refactor for other clients has been covered in PR #49763. After this and PR #49763 have been merged, I will double check that all clients have been correctly used with explicit version.

**Comment by kevin-wangzefeng** [comment]
kindly ping @lavalamp @madhusudancs

**Review by janetkuo — APPROVED** [review]

**Review by k82cn — APPROVED** [review]
LGTM

**Comment by kevin-wangzefeng** [comment]
Ping for approve, thanks.

**Comment by sttts** [comment]
@lavalamp this needs an approval.

**Comment by k82cn** [comment]
Should we get this one merged in 1.9?

**Comment by sttts** [comment]
> Should we get this one merged in 1.9?

yes, please. This must be rebased and we should get it in quickly then to avoid yet another rebase hell.

**Comment by kevin-wangzefeng** [comment]
@sttts thanks, I will rebase it today.

**Comment by sttts** [comment]
Rebase needed again.

**Comment by kevin-wangzefeng** [comment]
Refreshed, PTAL :)

**Comment by lavalamp** [comment]
/lgtm

In the future, doing this as a series of smaller changes will allow you to get better response time from leaf-level approvers/reviewers.

**Comment by k82cn** [comment]
rebase again :(.

**Comment by supereagle** [comment]
@lavalamp  There is a similar PR #49763 has been opened for a long time, could you help to approve? Thanks a lot :)

**Pull request merged**
